test_jpeg/jpegenc/subblocks/RLE/RLECore/rlecore.py

```diff
@@ -0,0 +1,179 @@
+""" core of the run length encoder module """
+
+from myhdl import always_comb, always_seq, block
+from myhdl import intbv, ResetSignal, Signal
+from myhdl.conversion import analyze
+from jpegenc.subblocks.RLE.RLECore.entropycoder import entropycoder
+
+
+WIDTH_RAM_ADDRESS = 6
+WIDTH_RAM_DATA = 12
+SIZE = 4
+
+class DataStream(object):
+    """ Input data streams into Rle Core """
+    def __init__(self, width=WIDTH_RAM_DATA):
+        self.start = Signal(bool(0))
+        self.input_val = Signal(intbv(0)[width:].signed())
+
+class RLESymbols(object):
+    """RLE symbols generatred by RLE Core"""
+    def __init__(self, width=WIDTH_RAM_DATA, size=SIZE):
+        self.runlength = Signal(intbv(0)[size:])
+        self.size = Signal(intbv(0)[size:])
+        self.amplitude = Signal(intbv(0)[width:].signed())
+        self.dovalid = Signal(bool(0))
+
+class RLEConfig(object):
+    """RLE configuration Signals are added here"""
```

Comment:
Most of the code in jpegenc (the little that exists) has used the style were class docstrings are under the `__init__`, we try and be consistent in the code (double check the style in that exists in the code).